src/pkg/utils/general_test.go

```
@@ -32,9 +41,79 @@ func TestContainsMention(t *testing.T) {
+}
+
+func (suite *UtilsTestSuite) TestStrToFloat64() {
+	assert.Equal(suite.T(), 123.45, utils.StrToFloat64("123.45"))
+	assert.Equal(suite.T(), 0.0, utils.StrToFloat64("invalid"))
+	assert.Equal(suite.T(), 0.0, utils.StrToFloat64(""))
+}
+
+func (suite *UtilsTestSuite) TestGetMetaDataFromURL() {
+	// Mock HTTP server
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(`<!DOCTYPE html><html><head><title>Test Title</title><meta name='description' content='Test Description'><meta property='og:image' content='http://example.com/image.jpg'></head><body></body></html>`))
```

golangci-lint (1.62.2)

Error return value of `w.Write` is not checked

(errcheck)